### ISSUE TITLE
Test test_tunnel_community_positive.py crashes

### ISSUE THREAD

**Issue by lfdversluis** [Tribler/tribler#1748]
Running it fails quite often, but also works from time to time.
I am running python 64 bit.

One of the crashes gives a segfault with no further info. Another crash gives the following stacktrace:

```
Running test_tunnel_community_positive.py...\n
test_anon_download (Tribler.Test.test_tunnel_community_positive.TestTunnelCommunityPositive) ... watchog wx thread *******TRIPPED!******, hasn't been set for 10.0174 secs.
disabling it and printing traces for all threads.:

*** STACKTRACE - START ***


### ThreadID: 140050508887808 Thread name: WatchDog
Locals by frame, innermost last:
/home/laurens/Documents/thesis/tribler/Tribler/Core/Utilities/instrumentation.py:172 print_all_stacks:
/home/laurens/Documents/thesis/tribler/Tribler/Core/Utilities/instrumentation.py:145 run:
/usr/lib/python2.7/threading.py:810 __bootstrap_inner:
/usr/lib/python2.7/threading.py:783 __bootstrap:

### ThreadID: 140050806220544 Thread name: Twisted
Locals by frame, innermost last:
/usr/lib/python2.7/dist-packages/twisted/internet/epollreactor.py:379 doPoll:
/usr/lib/python2.7/dist-packages/twisted/internet/base.py:1204 mainLoop:
/usr/lib/python2.7/dist-packages/twisted/internet/base.py:1192 run:
/home/laurens/Documents/thesis/tribler/Tribler/Core/Utilities/twisted_thread.py:55 _reactor_runner:
/usr/lib/python2.7/threading.py:763 run:
/usr/lib/python2.7/threading.py:810 __bootstrap_inner:
/usr/lib/python2.7/threading.py:783 __bootstrap:

### ThreadID: 140050871174976 Thread name: MainThread
Locals by frame, innermost last:
/usr/lib/python2.7/dist-packages/wx-2.8-gtk2-unicode/wx/_windows.py:703 ShowModal:
/home/laurens/Documents/thesis/tribler/Tribler/Main/tribler_main.py:378 InitStage1:
/home/laurens/Documents/thesis/tribler/Tribler/Main/tribler_main.py:134 __init__:
/home/laurens/Documents/thesis/tribler/Tribler/Main/tribler_main.py:1030 run:
/home/laurens/Documents/thesis/tribler/Tribler/Test/test_as_server.py:436 startTest:
/home/laurens/Documents/thesis/tribler/Tribler/Test/test_tunnel_base.py:110 startTest:
/home/laurens/Documents/thesis/tribler/Tribler/Test/test_tunnel_community_positive.py:94 test_anon_download:
/home/laurens/Documents/thesis/tribler/Tribler/Test/test_as_server.py:61 check:
/usr/lib/python2.7/unittest/case.py:331 run:
/usr/lib/python2.7/unittest/case.py:395 __call__:
/usr/lib/python2.7/dist-packages/nose/case.py:151 runTest:
/usr/lib/python2.7/dist-packages/nose/case.py:133 run:
/usr/lib/python2.7/dist-packages/nose/case.py:45 __call__:
/usr/lib/python2.7/dist-packages/nose/suite.py:223 run:
/usr/lib/python2.7/dist-packages/nose/suite.py:176 __call__:
/usr/lib/python2.7/dist-packages/nose/suite.py:223 run:
/usr/lib/python2.7/dist-packages/nose/suite.py:176 __call__:
/usr/lib/python2.7/dist-packages/nose/suite.py:223 run:
/usr/lib/python2.7/dist-packages/nose/suite.py:176 __call__:
/usr/lib/python2.7/dist-packages/nose/core.py:62 run:
/usr/lib/python2.7/dist-packages/nose/core.py:207 runTests:
/usr/lib/python2.7/unittest/main.py:95 __init__:
/usr/lib/python2.7/dist-packages/nose/core.py:121 __init__:
|          env = {'LC_NUMERIC': 'nl_NL.UTF-8', 'MANDATORY_PATH': '/usr/share/gconf/ubuntu.mandatory.path', 'XDG_GREETER_DATA_DIR': '/var/lib/lightdm-data/laurens', 'GNOME_DESKTOP_SESSION_ID': 'this-is-deprecated', 'UPSTART_EVENTS': 'started starting', 'LESSOPEN': '| /usr/bin/lesspipe %s', 'LC_PAPER': 'nl_NL.UTF-8', 'QT_IM_MODULE': 'ibus', 'LOGNAME': 'laurens', 'USER': 'laurens', 'PATH': '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/laurens/Documents/thesis/trible...
|       config = Config(addPaths=True, args=(), configSection='nosetests', debug='DEBUG', debugLog=None, env={}, exclude=None, files=[], firstPackageWins=False, getTestCaseNamesCompat=False, ignoreFiles=[<_sre.SRE_Pattern object at 0x7f602253e430>, <_sre.SRE_Pattern object at 0x7f602252c230>, <_sre.SRE_Pattern object at 0x7f602266ec30>], ignoreFilesDefaultStrings=['^\\.', '^_', '^setup\\.py$'], include=None, includeExe=False, logStream=<open file '<stderr>', mode 'w' at 0x7f60226b61e0>, loggingConfig=None, optio...
/usr/bin/nosetests:9 <module>:

*** STACKTRACE - END ***
```

Another crash:

```
Running test_tunnel_community_positive.py...
test_anon_download (Tribler.Test.test_tunnel_community_positive.TestTunnelCommunityPositive) ... python: ../nptl/pthread_mutex_lock.c:350: __pthread_mutex_lock_full: Assertion `(-(e)) != 3 || !robust' failed.
Aborted (core dumped)
```

**Comment by whirm** [comment]
The first one is telling you the wx thread got stuck for more than 10 seconds.
The second one you will have to run it with gdb and investigate.

**Comment by lfdversluis** [comment]
Should be fixed by PR #1757 and  #1756

**Comment by lfdversluis** [comment]
Fixed by #1800
